contributing/ansible-development.rst

```diff
@@ -0,0 +1,98 @@
+OME roles registered in https://github.com/ome/ansible-roles.
+
+Source code
+-----------
+
+The source code of a Java library should be maintained under version control
+using Git_ and hosted on GitHub_ under the
+`openmicroscopy <http://github.com/openmicroscopy/>`__ organization.
+Git repositories should be named as `ansible-role-<ROLE_NAME>`. Each directory
+layout should minimally follow the standard
+`Ansible role layout <https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-directory-structure>`_ including other files and folders for testing and
+deployment. A typical role structure is shown below::
+
+    defaults/           # Default variables
+    handlers/           # Handlers
+    meta/               # Role metadaa
+        main.yml        # Dependencies and Galaxy metadata
+    molecule/           # Test
+    tasks/              # Main list of tasks to be executed
+        main.yml
+    templates/          # Role templates
+    .travis.yml         # CI/deployment configuration file
+    README.md
+
+Versioning
+----------
+
+.. _PEP440: https://www.python.org/dev/peps/pep-0440/#semantic-versioning
+
+Ansible roles must follow the PEP440_ scheme for versioning. Final releases
+must also be compliant with `Semantic Versioning`_ i.e. a final version must
+be expressed as `MAJOR.MINOR.PATCH` where:
+
+- the MAJOR version must be incremented when incompatible API changes are made,
+- the MINOR version must be incremented when functionality is added in a
+  backwards-compatible manner, and
+- the PATCH version must be incremented when you backwards-compatible bug
+  fixes are made.
+
+Final releases must be tagged with a tag matching the version i.e. 
+`MAJOR.MINOR.PATCH` with no prefix for integration with Galaxy.
+
+Development releases or pre-releases must follow the PEP440_ scheme e.g.
+`MAJOR.MINOR.PATCH.dev1`, `MAJOR.MINOR.PATCHa1` `MAJOR.MINOR.PATHrc1`.
+
+Testing and Continuous Integration
+----------------------------------
+
+.. _Molecule: https://molecule.readthedocs.io/
+
+For each Ansible role, a :file:`molecule` folder should be configured allowing
+the testing to be tested using  Molecule_. One or multiple scenarios should be
+configured using at least a Docker driver if possible. A generic
+:file:`molecule` folder can be initialized using the following command::
+
+    molecule init scenario -r ansible-role-<NAME> -s default -d docker
+
+
+Continuous Integration of Ansible roles is performed using Travis CI.
+
+.. note::
+   OME Ansible roles are getting progressively upgraded from Molecule 1.x to 
+   Molecule 2.x. New roles should be configured using Molecule 2.x.
+
+Distribution and support
+------------------------
+
+All core OME Ansible roles should be deployed to
+`Ansible Galaxy <https://galaxy.ansible.com>`_ under the
+`openmicroscopy <https://galaxy.ansible.com/openmicroscopy/>`__ organization.
+All roles must support EL 7 as a primary platform. New roles should also
```

Comment:
EL 7? Is this the RedHat thing or what? Bit more context could help.

Reply:
Yes EL is largely RHEL/CentOS. I used this as this is the controlled vocabulary used in the Galaxy API (https://galaxy.ansible.com/api/v1/platforms/?name=EL) but I can switch to RHEL/CentOS for clarity.